apollo-runtime/src/main/java/com/apollographql/apollo/IdleCallback.java

```diff
@@ -0,0 +1,14 @@
+package com.apollographql.apollo;
+
+/**
+ * Callback which gets invoked when the {@link ApolloClient} transitions
+ * from active to idle state.
+ */
+public interface IdleCallback {
```

Comment:
Let's make it generic and call `IdleResourceCallback`.
Plus replace `{@link ApolloClient}` to `resource` in java docs